rust/stackable-cockpit/src/platform/service.rs

```diff
@@ -45,15 +45,54 @@ pub async fn get_endpoints(
     object_name: &str,
     object_namespace: &str,
 ) -> Result<IndexMap<String, String>, Error> {
-    let service_list_params =
+    let mut endpoints = IndexMap::new();
+    let list_params =
         ListParams::from_product(product_name, Some(object_name), k8s::ProductLabel::Name);
 
-    let services = kube_client
-        .list_services(Some(object_namespace), &service_list_params)
+    let listeners = kube_client
+        .list_listeners(Some(object_namespace), &list_params)
         .await
         .context(KubeClientFetchSnafu)?;
 
-    let mut endpoints = IndexMap::new();
+    for listener in &listeners {
+        let Some(listener_status) = &listener.status else {
+            continue;
+        };
+        for address in listener_status.ingress_addresses.iter().flatten() {
+            for port in &address.ports {
+                // Listener names usually have the pattern "listener-simple-hdfs-namenode-default-0" or
```

Comment:
suggestion: We should make the naming consistent. Then we could add a stable function which extracts the raw display name.

Reply:
The problem is that the ListenerClass name is equal to the name of the PVC that creates it.
And the pvc naming is done by k8s based on the name of the Pod and the name of the volume in the PodSpec.
And it also differs depend on wether a persistent or ephemeral volume is used :/ So everything is a mess....

Reply:
Well, that's just... great. I guess we can make it stable(ish) sometime in the future. Is it still worth pulling the code out into its own function?

Reply:
Sure, please have a look at 1a11e41

Reply:
Nice change, I like the addition of some test cases!

Reply:
Oh, huh. Guess I missed sending my reply here the other day (or GH ate it). I'd personally much rather just show the name as-is than mangle it like this. That would also make it easier/clearer for users to move between the "easy facade" of stackablectl and reading the backend data via kubectl when something goes wrong.